File: Lv.1/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    long answer = 0;
    long n=num;
    for(long i=1;i<=500;i++)
    {
        if(n==1)
        {
          break;   
        }
        
        else
        {
            if(n%2==0)
            {
                n/=2;
                answer++;
            }
            
            else
            {
                n=n*3+1;
                answer++;
            }
        }
    }
    
    if(n!=1)
    {
        answer= -1;
    }
    
    return answer;
}

File: Java/LV1/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int answer = 0;
        int counter = 0;
        long number = (long)num;
        while(true){
            
            if(number==1){
                answer=counter;
                break;
            }
            
            if(counter>=500){
                answer=-1;
                break;
            }
            
            if((number%2==0)){
                number/=2;
            }
            else{
                number*=3;
                number+=1;
            }
            
            counter++;
        }
        return answer;
    }
}
